pipenv_devcheck/pipenv_setup_comp.py

```diff
@@ -1,123 +1,126 @@
     """
+    Main wrapper around reading dependencies and running all checks
 
     Args:
         setup_filename (str, default "setup.py"):
             Location/name of file to be used as setup.py
         pipfile_filename (str, default "Pipfile"):
             Location/name of file to be used as Pipfile
     Returns:
-        tuple<str, str>:
+        tuple<dict<str, list<tuple<str, str>>>:
             Dictionaries of the dependencies found in setup.py and the Pipfile
     """
-    setup_lines, pipfile_lines = read_dep_files(setup_filename,
-                                                pipfile_filename)
+    setup_deps = get_setup_deps(setup_filename)
+    pipfile_deps = get_pipfile_deps(pipfile_filename)
+    run_checks(setup_deps, pipfile_deps)
+    return setup_deps, pipfile_deps
 
-    setup_deps_text = extract_deps_text(setup_lines, "setup")
-    pipfile_deps_text = extract_deps_text(pipfile_lines, "pipfile")
-    setup_deps = deps_text_to_dict(setup_deps_text, "setup")
-    pipfile_deps = deps_text_to_dict(pipfile_deps_text, "pipfile")
 
-    run_checks(setup_deps, pipfile_deps)
+def get_setup_deps(setup_filename="setup.py"):
+    """
+    Parses dependencies from specified setup file and
+    returns them as a dictionary
+
+    Args:
+        setup_filename (str, default "setup.py"):
+            Location/name of file to be used as setup.py
+    Returns:
+        tuple<dict<str, list<tuple<str, str>>>:
+            Dictionaries of the dependencies found in setup.py
+    """
+    setup_deps_str = read_setup(setup_filename)
 
+    setup_deps = {}
+    for dep in setup_deps_str:
+        parsed_dep = re.findall(setup_exp, dep)
+        if len(parsed_dep):
+            parsed_dep = parsed_dep[0]
+            setup_deps.update({parsed_dep[0]: [spec for spec in parsed_dep[1:]
+                                               if spec != ""]})
+        else:
+            setup_deps.update({dep: ["*"]})
 
-def read_dep_files(setup_filename, pipfile_filename):
+    setup_deps = split_ops_and_versions(setup_deps)
+    return setup_deps
+
+
+def read_setup(setup_filename):
     """
-    Reads text as an array of lines from files at specified names
     """
     with open(setup_filename, "r") as f:
-        setup_lines = f.readlines()
-
-    with open(pipfile_filename, "r") as f:
-        pipfile_lines = f.readlines()
+        setup_tree = ast.parse(f.read(), setup_filename)
+    for node in ast.walk(setup_tree):
+        is_setup_node = (hasattr(node, "func") and
+                         hasattr(node.func, "id") and
+                         node.func.id == "setup")
+        if is_setup_node:
+            setup_node = node
 
-    return setup_lines, pipfile_lines
+    for kw in setup_node.keywords:
+        if kw.arg == "install_requires":
+            setup_deps_str = ast.literal_eval(kw.value)
+            return setup_deps_str
 
 
-def extract_deps_text(lines, type):
+def get_pipfile_deps(pipfile_filename="Pipfile"):
```

Comment:
Can you call a Pipfile anything besides `Pipfile`?

Reply:
Realistically no - I put the argument in there for some early development easiness initially. The reason I've kept it in though is because I don't yet know a good way to mock out file I/O for testing. So in order to test `read_setup` and `read_pipfile`, I either need to be able to pass in different file names, or make `__main__.open()` behave differently within those functions. I could probably do a bit more research on this